distribution/lib/Standard/Base/0.0.0-dev/src/Data/Decimal.enso

```diff
@@ -0,0 +1,586 @@
+          Create a `Decimal` from a string.
+
+              d = Decimal.parse "12.345"
+    parse : Text -> Math_Context -> Decimal ! Arithmetic_Error | Number_Parse_Error
+    parse (s : Text) (mc : Math_Context | Nothing)=Nothing -> Decimal | Number_Parse_Error =
+        handle_java_exception <| handle_number_format_exception <|
+            case mc of
+                _ : Math_Context -> Decimal.Value <| handle_precision_loss s <| Decimal_Utils.fromString s mc.math_context
+                _ : Nothing -> Decimal.Value (Decimal_Utils.fromString s)
+
+    ## GROUP conversions
+       Construct a `Decimal` from an `Integer`.
+
+       Arguments:
+       - i: The `Integer` to construct a `Decimal` from.
+       - mc: The `Math_Context` to use to specify precision and `Rounding_Mode`.
+         If a `Math_Context` is used, there is a possibility of a loss of
+         precision.
+
+       ! Error Conditions
+
+         - If the construction of the Decimal results in a loss of precision, a
+           `Loss_Of_Numeric_Precision` warning is attached. This can only happen
+           if a `Math_Context` value is explicitly passed.
+
+        ^ Example
+          Create a `Decimal` from an integer.
+
+              d = Decimal.from_integer 12
+    from_integer : Integer -> Math_Context -> Decimal ! Arithmetic_Error
+    from_integer (i : Integer) (mc : Math_Context | Nothing)=Nothing -> Decimal ! Arithmetic_Error =
+        handle_java_exception <|
+            case mc of
+                _ : Math_Context -> Decimal.Value <| handle_precision_loss i <| Decimal_Utils.fromInteger i mc.math_context
+                _ : Nothing -> Decimal.Value (Decimal_Utils.fromInteger i)
+
+    ## GROUP conversions
+       Construct a `Decimal` from a `Float`.
+
+       Arguments:
+       - f: The `Float` to construct a `Decimal` from.
+       - mc: The `Math_Context` to use to specify precision and `Rounding_Mode`.
+         If a `Math_Context` is used, there is a possibility of a loss of
+         precision.
+
+       ? Precision Loss
+
+         A `Decimal` can represent any `Float` precisely. However, some `Float`
+         literals, such as `0.1`, cannot be represented exactly to infinite
+         precision by a `Float`. For this reason, constructing a `Decimal` from
+         a `Float` always attaches a `Loss_Of_Numeric_Precision` warning to the
+         result.
+
+       ! Error Conditions
+
+         - A `Loss_Of_Numeric_Precision` warning is always attached when
+           converting to `Decimal` from `Float`.
+         - If `f` is NaN or +/-Inf, an Illegal_Argument error is thrown.
+
+        ^ Example
+          Create a `Decimal` from a float.
+
+              d = Decimal.from_integer 12.345
+    from_float : Float -> Math_Context -> Decimal ! Arithmetic_Error ! Illegal_Argument
+    from_float (f : Float) (mc : Math_Context | Nothing)=Nothing -> Decimal ! Arithmetic_Error ! Illegal_Argument =
+        is_exceptional = f.is_nan || f.is_infinite
```

Comment:
I think we can simplify this to:

```suggestion
        is_exceptional = f.is_finite.not
```

Reply:
I think that might read as "is infinite" unless the developer remembers that NaN is not finite. NaN behavior is quite variable across implementations (both software and hardware) so I wouldn't want to rely on that assumption.